detector.go

```diff
@@ -21,6 +21,7 @@ const (
 )
 
 var ErrFail = errors.New("no buildpacks participating")
+var ErrFailWithErrors = errors.New("no buildpacks participating: failed with errors")
```

Comment:
Maybe `ErrBuildpack` instead of `ErrFailWithErrors` to make clear that the error originated from the Buildpack?

I know this was already here but I also think a less generic name like`ErrFailedDetection` might be clearer than `ErrorFail` given that this variable lives in a package with many other phases.